tests/tracking/test_fluent.py

```diff
@@ -401,29 +402,107 @@ def test_search_runs_no_arguments():
+    max_results = 50
+    with mock.patch("mlflow.tracking.fluent.NUM_RUNS_PER_PAGE_PANDAS", 10):
+        with mock.patch.object(MlflowClient, "search_runs"):
+            MlflowClient.search_runs.side_effect = [tokenized_runs, tokenized_runs, no_token_runs]
+            TOTAL_RUNS = 21
+
+            paginated_runs = _get_paginated_runs([], "", ViewType.ACTIVE_ONLY, max_results, None)
+            assert len(paginated_runs) == TOTAL_RUNS
+
+
+def test_get_paginated_runs_eq_maxresults_blanktoken():
+    # runs returned equal to max_results, blank token
+    runs = [create_run() for i in range(10)]
+    tokenized_runs = PagedList(runs, "")
+    no_token_runs = PagedList([], "")
+    max_results = 10
+    with mock.patch("mlflow.tracking.fluent.NUM_RUNS_PER_PAGE_PANDAS", 10):
+        with mock.patch.object(MlflowClient, "search_runs"):
+            MlflowClient.search_runs.side_effect = [tokenized_runs, no_token_runs]
+            paginated_runs = _get_paginated_runs([], "", ViewType.ACTIVE_ONLY, max_results, None)
+            MlflowClient.search_runs.assert_called_once()
+            assert len(paginated_runs) == 10
+
+
+def test_get_paginated_runs_eq_maxresults_token():
+    # runs returned equal to max_results, token appended which if used will result in no runs
+    runs = [create_run() for i in range(10)]
+    tokenized_runs = PagedList(runs, "abc")
+    blank_runs = PagedList([], "")
+    max_results = 10
+    with mock.patch("mlflow.tracking.fluent.NUM_RUNS_PER_PAGE_PANDAS", 10):
+        with mock.patch.object(MlflowClient, "search_runs"):
+            MlflowClient.search_runs.side_effect = [tokenized_runs, blank_runs]
+            paginated_runs = _get_paginated_runs([], "", ViewType.ACTIVE_ONLY, max_results, None)
+            MlflowClient.search_runs.assert_called_once()
+            assert len(paginated_runs) == 10
+
+
+def test_get_paginated_runs_gt_maxresults():
+    # should ask for and return the correct number of max_results
+    full_page_runs = PagedList([create_run() for i in range(8)], "abc")
+    partial_page = PagedList([create_run() for i in range(4)], "def")
+    max_results = 20
+    with mock.patch("mlflow.tracking.fluent.NUM_RUNS_PER_PAGE_PANDAS", 8):
+        with mock.patch.object(MlflowClient, "search_runs"):
+            MlflowClient.search_runs.side_effect = [full_page_runs, full_page_runs, partial_page]
+            paginated_runs = _get_paginated_runs([12], "", ViewType.ACTIVE_ONLY, max_results, None)
+            calls = [mock.call([12], "", ViewType.ACTIVE_ONLY, 8, None, None),
+                     mock.call([12], "", ViewType.ACTIVE_ONLY, 8, None, "abc"),
+                     mock.call([12], "", ViewType.ACTIVE_ONLY, 20 % 8, None, "abc")]
+            MlflowClient.search_runs.assert_has_calls(calls)
+            assert len(paginated_runs) == 20
+
+
+def test_get_paginated_maxresults_lt_runs_per_page():
+    # should only get max_result number of runs, should only call search_runs once
```

Comment:
What's the reason we don't combine this with `test_get_paginated_runs_lt_maxresults_notoken`?

Reply:
Got rid of the `notoken` tests. This should now have a more descriptive test docstring